src/core/Carousel/CarouselSlider.tsx

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { CarouselContext } from './CarouselContext';
+import { useMergedRefs } from '../utils';
+
+/**
+ * `CarouselSlider` is the scrollable list that should consist of `CarouselSlide` components.
+ */
+export const CarouselSlider = React.forwardRef<
+  HTMLOListElement,
+  React.ComponentPropsWithoutRef<'ol'>
+>((props, ref) => {
+  const { children, className, ...rest } = props;
+  const context = React.useContext(CarouselContext);
+
+  if (!context) {
+    throw new Error('CarouselSlider must be used within Carousel');
+  }
+
+  const { currentIndex, setSlideCount, idPrefix } = context;
+
+  const items = React.useMemo(
+    () =>
+      React.Children.map(children, (child, index) =>
+        React.isValidElement(child)
+          ? React.cloneElement(child, {
+              id: `${idPrefix}--slide-${index}`,
+              index,
+            })
+          : child,
+      ) ?? [],
+    [children, idPrefix],
+  ) as React.ReactNode[];
+
+  React.useLayoutEffect(() => {
+    setSlideCount(items.length);
+  }, [items, setSlideCount]);
+
+  const [mountState, setMountState] = React.useState<
+    'not-mounted' | 'just-mounted' | 'post-mount'
+  >('not-mounted');
+
+  React.useLayoutEffect(() => {
+    if (mountState === 'not-mounted') {
+      setMountState('just-mounted');
+    } else if (mountState === 'just-mounted') {
+      setMountState('post-mount');
+    }
+  }, [mountState]);
```

Comment:
Need opinions on this prop. I wanted a triple-state value, so I went with a string union. Another option would be to use `boolean | undefined` or `boolean | 'not-mounted'`.

Reply:
I like the string union. An alternative `boolean | 'just-mounted'` could work instead of `boolean | 'not-mounted'`.